src/ui/tabs/rooms_tab.py

```
@@ -190,49 +206,55 @@ def calculate_rooms(self):
-                            raise ValueError(f"{field_name} cannot be negative in Room {i+1}: {value}")
-                        return value
+                            raise
+
+                try:
+                    present_unit = _to_int_safe(present_text)
+                    previous_unit = _to_int_safe(previous_text)
+                except ValueError:
+                    raise ValueError(
+                        f"Non-numeric input in Room {i+1}. "
+                        f"Present: '{present_text}', Previous: '{previous_text}'"
+                    )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)